### PR TITLE
Ammo cleanup

### PR DESCRIPTION
**Pull request by pubby**
Creates a one-to-one mapping between casing and bullet types to make future changes easier. No longer will we have to deal with `weakbullet3` being shared across multiple ammunitions.

Cleans up ammo definitions to make them easier to read and maintain. It's far from perfect, but its a start.

Removes the gang, haem, and toxic bullets for being unnecessary. Some of you might be thinking: "Those sound cool!!! Keep them!!!" But in reality they were just renamed regular bullets that dealt less damage.
 
Combines meteorshot bullets with breaching bullet. These were practically the same anyways.

Removes overloaded improvised shells because people complain that they are broken. Someone can readd these when they figure out how to make them not broken. The regular improvised shells remain.

### CODE REVIEW

**Comment by PraiseRatvar** [comment]
i hope nothing breaks

**Comment by pubby** [comment]
breaks Jmad's heart

**Comment by Bawhoppen** [comment]
This doesn't improve anything, and just removes more depth from bullets

**Comment by pubby** [comment]
Nothing of value was lost.

Toxic bullets were just a lazy varedit of regular bullets that dealt instant TOX damage rather than BRUTE. There is zero point in keeping them around when we have reagent bullets/darts/syringes instead. Again, we have fucking REAGENT BULLETS. We do not need to keep around lazy old crap when we have new and better stuff.

Gang bullets were removed because gang mode was removed.

Haem bullets were less effective at killing people than regular bullets, and thus they were nothing more than a beginners trap. If you want to make people bleed: use hollow-point. If you want to stun people: use stunning bullets. See WJohn's post:

> Haem is terrible. A human starts with 560 units of blood (can be overfilled to 600 through injections with no negative effects), and will die at around 60% of their blood contents, so around ~340/560 blood. Now you CAN make them unconscious without killing them at a higher value, like at around maybe 400. The problem is that haem rounds from the l6 saw only deal 20 brute and 35 blood removal, so you'd have to shoot them a ton for them to VERY SLOWLY start suffocating to death due to lack of blood, when you could just use HP rounds and kill anyone in 2-3 shots, depending on if they have armor or not. Sniper rounds also fire much weaker killing projectiles that remove 100 blood. Again, just use the standard rounds that knocks them down long enough to hit them again, or use xray to hit them behind walls where they have no way to see this coming and dodge it (great gameplay there).

Breaching shells were just renamed meteor shells with slightly less damage. There was no difference otherwise, so it makes sense to consolidate the two.

Improvised overloaded was removed because we have craftable frag bullets which are the same thing except actually balanced.

**Comment by InsaneHyena** [comment]
Can you hear the screeching of ballisticfags?

File: code/modules/projectiles/projectile/special.dm [hunk 1]
```diff
@@ -616,3 +595,19 @@
...
+	name = "neurotoxin spit"
+	icon_state = "neurotoxin"
+	damage = 5
+	damage_type = TOX
+	knockdown = 100
+
+/obj/item/projectile/bullet/neurotoxin/on_hit(atom/target, blocked = FALSE)
+	if(isalien(target))
+		knockdown = 0
+		nodamage = 1
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/special.dm]: TRUE

File: code/modules/projectiles/projectile/bullets.dm [hunk 2]
```diff
@@ -8,100 +8,233 @@
...
+
+/obj/item/projectile/bullet/incendiary/on_hit(atom/target, blocked = FALSE)
+	. = ..()
+	if(iscarbon(target))
+		var/mob/living/carbon/M = target
+		M.adjust_fire_stacks(fire_stacks)
+		M.IgniteMob()
+
+/obj/item/projectile/bullet/incendiary/Move()
+	..()
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: . = ..()

File: code/modules/projectiles/projectile/bullets.dm [hunk 3]
```diff
@@ -8,100 +8,233 @@
...
+	icon_state= "bolter"
+	damage = 60
 
-/obj/item/projectile/bullet/pellet/overload/on_range()
-	explosion(src, 0, 0, 2)
-	do_sparks(3, TRUE, src)
+/obj/item/projectile/bullet/a40mm/on_hit(atom/target, blocked = FALSE)
 	..()
+	explosion(target, -1, 0, 2, 1, 0, flame_range = 3)
+	return 1
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: Should this be TRUE?

File: code/modules/projectiles/projectile/bullets.dm [hunk 4]
```diff
@@ -8,100 +8,233 @@
...
+	if((blocked != 100) && (!ismob(target) && breakthings))
+		target.ex_act(rand(1,2))
+	return ..()
 
-/obj/item/projectile/bullet/midbullet3
-	damage = 30
+/obj/item/projectile/bullet/p50/soporific
+	name =".50 soporific bullet"
+	armour_penetration = 0
+	nodamage = 1
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: TRUE

File: code/modules/projectiles/projectile/bullets.dm [hunk 5]
```diff
@@ -8,100 +8,233 @@
...
+	dismemberment = 0
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: FALSE

File: code/modules/projectiles/projectile/bullets.dm [hunk 6]
```diff
@@ -8,100 +8,233 @@
...
+	knockdown = 0
+	breakthings = FALSE
+
+/obj/item/projectile/bullet/p50/soporific/on_hit(atom/target, blocked = FALSE)
+	if((blocked != 100) && isliving(target))
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: This implies blocked is a bool and an int?

File: code/modules/projectiles/projectile/bullets.dm [hunk 7]
```diff
@@ -8,100 +8,233 @@
...
+	damage = 70
+	knockdown = 100
+	dismemberment = 50
+	armour_penetration = 50
+	var/breakthings = TRUE
 
-/obj/item/projectile/bullet/midbullet2
-	damage = 25
+/obj/item/projectile/bullet/p50/on_hit(atom/target, blocked = FALSE)
+	if((blocked != 100) && (!ismob(target) && breakthings))
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: This implies blocked is a bool and an int?

File: code/modules/projectiles/projectile/bullets.dm [hunk 8]
```diff
@@ -8,100 +8,233 @@
...
+		L.Sleeping(400)
+	return ..()
 
-/obj/item/projectile/bullet/midbullet3/hp
+/obj/item/projectile/bullet/p50/penetrator
+	name =".50 penetrator bullet"
+	icon_state = "gauss"
+	name = "penetrator round"
+	damage = 60
+	forcedodge = 1
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: TRUE

File: code/modules/projectiles/projectile/bullets.dm [hunk 9]
```diff
@@ -8,100 +8,233 @@
...
+	dismemberment = 0 //It goes through you cleanly.
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: FALSE

File: code/modules/projectiles/projectile/bullets.dm [hunk 10]
```diff
@@ -110,41 +243,84 @@
...
+	damage = 25
+	knockdown = 50
 
-/obj/item/projectile/bullet/incendiary/shell
-	name = "incendiary slug"
-	damage = 20
+/obj/item/projectile/bullet/shotgun_frag12/on_hit(atom/target, blocked = FALSE)
+	..()
+	explosion(target, -1, 0, 1)
+	return 1
```
Comment (Cyberboss) [inline code/modules/projectiles/projectile/bullets.dm]: TRUE?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile/bullets.dm (hunk 5 above)
Comment (pubby) [reply code/modules/projectiles/projectile/bullets.dm]: Which are you talking about? `armour_penetration` an `dismemberment` are integers, I think.

**Comment by Tacolizard** [comment]
good idea, but this is the wrong way of going about it. 

**if bullets and casings are now 1-1, then you need to refactor guncode and remove bullets altogether. have damage and all their vars instead defined on the casing. there is no need for having these pieces seperate.**

**Comment by pubby** [comment]
The separation between bullets and casings seems logical to me. I don't see how that would improve anything. And instead of proc overloads there'd be huge switch statement garbage.

**Comment by RemieRichards** [comment]
Ignore taco.

**Pull request merged**